### ISSUE TITLE
T.RandomRotation in custom dataloader yields strange results 

### ISSUE THREAD

**Issue by ItamarSafriel** [facebookresearch/detectron2#1799]
Hello,
I have trained three models on a custom dataset (same dataset every time, the training dataset has 800 images). 
Each model has converged and was trained with a different angle given to ```T.RandomRotation```. The angles were 5, 10 and 15 degrees. 
Here are some examples for inferences made with each model on the same image from the test set: 

![all_models](https://user-images.githubusercontent.com/49122015/88029716-06780e00-cb43-11ea-8dc5-0b0aa8635b06.png)

And here are inferences with the same models, but when the images are rotated with respect to the training angle:
![all_models_rotated](https://user-images.githubusercontent.com/49122015/88029268-75089c00-cb42-11ea-8414-11b1777ac6d3.png)

These results seem very strange. Firstly, I wouldn't expect that such a small change in the training angle would have such a big impact on the final model. Secondly, it doesn't make sense that rotating the input image would yield such a big improvement. It seems that instead of rotating each image to a random angle sampled with respect to the training angle, it is rotating all images to the same angle given by the training angle. Any ideas for what when wrong? maybe I need to use ```expand = True``` for T.RandomRotation? (though I am not sure how that might help). 

This is the code used for inference:
```python 
cfg = get_cfg()
cfg.MODEL.DEVICE = device
cfg.merge_from_file(model_zoo.get_config_file(model_name))
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1
cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.5
cfg.MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS = num_keypoints
cfg.INPUT.MIN_SIZE_TEST = 0
cfg.INPUT.CROP.ENABLED = True
cfg.INPUT.CROP.TYPE = "absolute"
cfg.INPUT.CROP.SIZE = [180, 270]
cfg.MODEL.WEIGHTS = path_to_config
predictor = DefaultPredictor(cfg)
image = plt.imread(image_path)
image = cv2.resize(image, (300, 200), interpolation=cv2.INTER_CUBIC)
outputs = predictor(image)
```

And this is the training code:
```python 
cfg = get_cfg()
cfg.MODEL.DEVICE = f"cuda:{device}" if torch.cuda.is_available() else "cpu"
cfg.merge_from_file(model_zoo.get_config_file(model_name))
cfg.DATASETS.TRAIN = (f"{bed_type}_Train",)
cfg.DATASETS.TEST = (f"{bed_type}_Test",)
cfg.DATALOADER.NUM_WORKERS = 2
cfg.OUTPUT_DIR = output_dir
cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(model_name)
cfg.SOLVER.BASE_LR = 0.00025
cfg.SOLVER.MAX_ITER = int(num_iter) 
cfg.SOLVER.IMS_PER_BATCH = int(batch_size)
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1 
cfg.MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS = 16
os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)

transfrom_list = [T.Resize(shape=(200,300)), 
                        T.RandomRotation(angle=angle, interp=Image.BILINEAR),
                        T.RandomContrast(intensity_min=0.75, intensity_max=1.25),
                        T.RandomBrightness(intensity_min=0.75, intensity_max=1.25),
                        T.RandomSaturation(intensity_min=0.75, intensity_max=1.25),
                        T.RandomLighting(scale=0.1),
                        T.RandomCrop(crop_type="absolute", crop_size=(180, 270))]

custom_mapper = get_custom_mapper(transfrom_list)

class CustomTrainer(DefaultTrainer):
    @classmethod
    def build_test_loader(cls, cfg: CfgNode, dataset_name):
        return build_detection_test_loader(cfg, dataset_name, mapper=custom_mapper)

    @classmethod
    def build_train_loader(cls, cfg: CfgNode):
        return build_detection_train_loader(cfg, mapper=custom_mapper)
trainer = CustomTrainer(cfg) 
trainer.resume_or_load(resume=False)
trainer.train()
```
1.full code you wrote or full changes you made (git diff)
git diff is empty (I did not change anything).
2.what exact command you run:
A custom command which runs inference with a model (same as notebook, code above).  

## Expected behavior:
As mentioned above.  

## Environment:

```
---------------------  ------------------------------------------------------------------------------------------------
sys.platform           linux
Python                 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31) [GCC 7.3.0]
numpy                  1.17.4
detectron2             0.1.3 @/home/itamar/projects/measure-kid-server/detectron2/detectron2
Compiler               GCC 7.5
CUDA compiler          not available
DETECTRON2_ENV_MODULE  <not set>
PyTorch                1.5.0 @/home/itamar/anaconda3/envs/measure-kid/lib/python3.6/site-packages/torch
PyTorch debug build    False
GPU available          False
Pillow                 6.2.1
torchvision            0.6.0a0+82fd1c8 @/home/itamar/anaconda3/envs/measure-kid/lib/python3.6/site-packages/torchvision
fvcore                 0.1.1.post20200615
cv2                    4.1.1
---------------------  ------------------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.4 Product Build 20190411 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=0, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

```

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.


If your question is about this:
>  It seems that instead of rotating each image to a random angle sampled with respect to the training angle, it is rotating all images to the same angle given by the training angle

you might double-check the documentation https://detectron2.readthedocs.io/modules/data.html#detectron2.data.transforms.RandomRotation and make sure you're using a list of angles as the documentation suggests.
